prober/http_test.go

```diff
@@ -97,6 +100,140 @@ func TestValidHTTPVersion(t *testing.T) {
 	}
 }
 
+func TestContentLength(t *testing.T) {
+	type testdata struct {
+		msg                    string
+		contentLenght          int
+		uncompressedBodyLength int
+		handler                http.HandlerFunc
+	}
+
+	// These test cases cover known encodings (deflate, gzip,
+	// identity) so changes in the underlying libraries don't cause
+	// undetected changes in behavior.
+	//
+	// They also cover unknown encodings (xxx) to verify that those
+	// are handled correctly.
+	testcases := map[string]testdata{
+		"deflate": func() testdata {
+			msg := strings.Repeat("hello world", 10)
+			var buf bytes.Buffer
+			// the only error path is an invalid compression level
```

Comment:
Comments should be full sentences with capital letters and full stops.

Reply:
Thanks.

This part of the code was removed because of the other comments.